File: src/main/java/hu/elte/alkfejl/cheaptrade/domain/bid/Bid.java
package hu.elte.alkfejl.cheaptrade.domain.bid;

import hu.elte.alkfejl.cheaptrade.domain.base.BaseEntity;
import hu.elte.alkfejl.cheaptrade.domain.item.Item;
import hu.elte.alkfejl.cheaptrade.domain.user.User;
import lombok.*;

import javax.persistence.Entity;
import javax.persistence.ManyToOne;
import java.math.BigDecimal;


@NoArgsConstructor
@AllArgsConstructor
@Builder
@Getter
@Setter
@ToString(callSuper = true)

@Entity
public class Bid extends BaseEntity {

    @ManyToOne
    private User user;

    @ManyToOne
    private Item item;

    private BigDecimal amount;

    private boolean isBuyout;

}

File: src/main/client/src/app/model/bid.ts
import {User} from './User';
import {Item} from './Item';
import {Observable} from "rxjs";


export class Bid {
  id: number;
  createdAt: string;
  user: Observable<User>;
  item: Item
  amount: number;
  isBuyout: boolean;

  constructor(user: Observable<User>, item: Item, amount: number) {

  }
}
